### ISSUE TITLE
Compiling and connecting via Serial Monitor

### ISSUE THREAD

**Issue by dcabro** [DesconBelgrade/Klimerko#12]
Executable segment sizes:
IROM   : 284884          - code in flash         (default or ICACHE_FLASH_ATTR) 
IRAM   : 30608   / 32768 - code in IRAM          (ICACHE_RAM_ATTR, ISRs...) 
DATA   : 1492  )         - initialized variables (global, static) in RAM/HEAP 
RODATA : 7948  ) / 81920 - constants             (global, static) in RAM/HEAP 
BSS    : 31136 )         - zeroed variables      (global, static) in RAM/HEAP 
Sketch uses 324932 bytes (31%) of program storage space. Maximum is 1044464 bytes.
Global variables use 40576 bytes (49%) of dynamic memory, leaving 41344 bytes for local variables. Maximum is 81920 bytes.
esptool.py v2.8
Serial port COM3
Connecting........_____....._____....._____....._____....._____....._____.....____Traceback (most recent call last):
  File "C:\Users\Darko\AppData\Local\Arduino15\packages\esp8266\hardware\esp8266\2.7.4/tools/upload.py", line 65, in <module>
    esptool.main(cmdline)
  File "C:/Users/Darko/AppData/Local/Arduino15/packages/esp8266/hardware/esp8266/2.7.4/tools/esptool\esptool.py", line 2890, in main
    esp.connect(args.before)
  File "C:/Users/Darko/AppData/Local/Arduino15/packages/esp8266/hardware/esp8266/2.7.4/tools/esptool\esptool.py", line 483, in connect
    raise FatalError('Failed to connect to %s: %s' % (self.CHIP_NAME, last_error))
esptool.FatalError: Failed to connect to ESP8266: Timed out waiting for packet header
esptool.FatalError: Failed to connect to ESP8266: Timed out waiting for packet header

_
Also Serial monitor shows garbage although baud rate is properly set:

![serial-monitor](https://user-images.githubusercontent.com/3135899/98370508-f5440600-203a-11eb-86ba-736a2b0a4f4c.png)

**Comment by dcabro** [comment]
When pressing reset, Node MCU blinks the blue led once, and serial monitor shows that garbage. All software up to date.

**Comment by epiller** [comment]
Are you sure you're using the latest Arduino version, 1.8.13?

**Comment by dcabro** [comment]
Yes, 100% sure.
![arduino-nodemc-esp8266](https://user-images.githubusercontent.com/3135899/98672563-4a8f5880-2356-11eb-9906-05f40aa01c46.png)
![arduino-version-baud](https://user-images.githubusercontent.com/3135899/98672565-4b27ef00-2356-11eb-951f-6fca9df0bd83.png)

**Comment by epiller** [comment]
- Do you have any other Serial ports available?
- What does the back of your NodeMCU board say? It should be marked as "Amica"

**Comment by dcabro** [comment]
No,only one, COM3
Not sure which is the "back" side, but one side (where the pins are) is white, and has nothing inscribed 
Other side, with wifi module on it, has "HW-306" in the middle. Don't see "Amica" anywhere. Here is the pic. 

Btw, NodeMCU board was ordered months ago - I followed the link to AE from this project on github (btw it's broken right now). Looking at my order details, the item description was "V3 Wireless module NodeMcu 4M bytes Lua WIFI Internet of Things development board based ESP8266 ESP-12E for arduino CP2102"

**Comment by dcabro** [comment]
![IMG_20201110_161856](https://user-images.githubusercontent.com/3135899/98693348-7586a600-2370-11eb-89cd-4c1b4cfae5a2.jpg)

**Comment by epiller** [comment]
That's really weird. The board should look exactly like this: 
![NodeMCU upper side](https://upload.wikimedia.org/wikipedia/commons/e/e5/Nodemcu_amica_bot_02.png)
and like this on the bottom side
![NodeMCU upper side](http://s2.electrodragon.com/wp-content/uploads/2015/04/NodeMCU-LUA-Amica-R2-03.jpg)  

It seems like you've received a counterfeit NodeMCU for some reason.
- Try lowering the upload baud rate (Tools > Upload Speed) to 9600 and then try to upload the firmware again.
- The gibberish you're seeing in the Serial monitor is probably intelligible once you match its baud rate, so try scrolling through different baud rates in Serial monitor and see if the text starts making sense at some point (could be 9600 or 57600) and post it here if you get any results.

**Comment by dcabro** [comment]
Yeah, mine is different from that bottom side, but like I said, the exact one I ordered was linked to your page... 

Anyway, 74880 baud seems to be giving something readable, but it does not progress from here:

 ets Jan  8 2013,rst cause:2, boot mode:(7,2)

**Comment by epiller** [comment]
Can you try using 9600 baud rate to upload the firmware onto the device?
If that fails, keep the upload speed at 9600 but disconnect the NodeMCU from your computer, then press and hold the FLASH button on the NodeMCU and then (while holding the button), connect it to your computer and only then release the button, and then try uploading again at the same baud rate.

**Comment by dcabro** [comment]
57600 is the lowest speed I have in "upload speed"

**Comment by epiller** [comment]
Ah, right, sorry about that. Keep it at 115200 and try the second thing from the previous message.

**Comment by dcabro** [comment]
Arduino: 1.8.13 (Windows 10), Board: "NodeMCU 1.0 (ESP-12E Module), 80 MHz, Flash, Legacy (new can return nullptr), All SSL ciphers (most compatible), 4MB (FS:2MB OTA:~1019KB), 2, v2 Lower Memory, Disabled, None, Only Sketch, 115200"

Executable segment sizes:

IROM   : 284884          - code in flash         (default or ICACHE_FLASH_ATTR) 

IRAM   : 30608   / 32768 - code in IRAM          (ICACHE_RAM_ATTR, ISRs...) 

DATA   : 1492  )         - initialized variables (global, static) in RAM/HEAP 

RODATA : 7948  ) / 81920 - constants             (global, static) in RAM/HEAP 

BSS    : 31136 )         - zeroed variables      (global, static) in RAM/HEAP 

Sketch uses 324932 bytes (31%) of program storage space. Maximum is 1044464 bytes.

Global variables use 40576 bytes (49%) of dynamic memory, leaving 41344 bytes for local variables. Maximum is 81920 bytes.

esptool.py v2.8

Serial port COM3

Connecting........_____....._____....._____....._____....._____....._____.....____Traceback (most recent call last):

  File "C:\Users\darko\AppData\Local\Arduino15\packages\esp8266\hardware\esp8266\2.7.4/tools/upload.py", line 65, in <module>

    esptool.main(cmdline)

  File "C:/Users/darko/AppData/Local/Arduino15/packages/esp8266/hardware/esp8266/2.7.4/tools/esptool\esptool.py", line 2890, in main

    esp.connect(args.before)

  File "C:/Users/darko/AppData/Local/Arduino15/packages/esp8266/hardware/esp8266/2.7.4/tools/esptool\esptool.py", line 483, in connect

    raise FatalError('Failed to connect to %s: %s' % (self.CHIP_NAME, last_error))

esptool.FatalError: Failed to connect to ESP8266: Timed out waiting for packet header

esptool.FatalError: Failed to connect to ESP8266: Timed out waiting for packet header



_



This report would have more information with
"Show verbose output during compilation"
option enabled in File -> Preferences.

**Comment by dcabro** [comment]
And the serial console is showing: 

```
 ets Jan  8 2013,rst cause:2, boot mode:(7,6)

waiting for host
```

**Comment by dcabro** [comment]
bump

**Comment by dcabro** [comment]
In most of the cases, it's the STUPID user (that would be me), and not the hardware or the code...

Anyway, I ordered a new node mcu, and when it arrived, before I re-soldered everything, I wanted to make sure that it was okay in the first place. And you know what? PMS7003 RX was soldered do D8 instead of D6. So when I fixed that, everything started working without issues. 

So sorry for all the trouble and thank you!
